packages/client/lib/blockchain/chain.ts

```diff
@@ -221,17 +253,27 @@ export class Chain {
 
     const headers: ChainHeaders = {
       latest: null,
+      finalized: null,
+      safe: null,
       td: BigInt(0),
       height: BigInt(0),
     }
     const blocks: ChainBlocks = {
       latest: null,
+      finalized: null,
+      safe: null,
       td: BigInt(0),
       height: BigInt(0),
     }
 
     headers.latest = await this.getCanonicalHeadHeader()
+    // finalized and safe have to block only since execution has to be valid
```

Comment:
This comment is a little bit unclear.  Should it say something like `finalized and safe blocks are equivalent to canonical header since block has been executed`?